### PR TITLE
Leaderboards

### PR DESCRIPTION
**Pull request by afieldi**
Stats on frontend now are weighted when merging multiple rows. 

Some misc improvements

### CODE REVIEW

File: Common/utils.ts [hunk 1]
```diff
@@ -224,184 +224,9 @@ export function ObjectArrayToCsv(objects: { [key: string]: any }[], headers?: st
...
-  object1.totalGoldOfTeam += NonNone(object2.totalGoldOfTeam, 0);
-  object1.totalPhysicalDamageDealtToChampionsOfTeam += NonNone(object2.totalPhysicalDamageDealtToChampionsOfTeam, 0);
-  object1.totalMagicDamageDealtToChampionsOfTeam += NonNone(object2.totalMagicDamageDealtToChampionsOfTeam, 0);
-  object1.totalTrueDamageDealtToChampionsOfTeam += NonNone(object2.totalTrueDamageDealtToChampionsOfTeam, 0);
-  object1.totalDamageDealtToChampionsOfTeam += NonNone(object2.totalDamageDealtToChampionsOfTeam, 0);
-  object1.totalVisionScoreOfTeam += NonNone(object2.totalVisionScoreOfTeam, 0);
+  for (const key in object1) {
+    // @ts-ignore Typescript being stupid here. Just let me do this
+    object1[key] += NonNone(object2[key], 0);
+  }
```
Comment (Vexrax) [inline Common/utils.ts:230]: Very Very Very Cool

Comment (Vexrax) [reply Common/utils.ts:230]: But also what happens if a field in object1 is undefined? It shouldnt be possible but just curious.

File: front-end/src/components/player-page/stats/performance-overview.tsx [hunk 2]
```diff
@@ -53,21 +53,17 @@ function getStatBox(index: number, statGenerator: BaseStatGenerator, performance
...
+        sorted
+    );
 
     return <StatBox key={index}
                     statToolTip={statGenerator.getToolTip()}
                     statValue={statGenerator.getStatString(performanceOverviewProps.playerStats)}
                     statTitle={statGenerator.getStatTitle()}
                     haveStatsLoaded={statGenerator.canLoadData(performanceOverviewProps.playerStats)}
-                    shouldShowLeaderboard={isPlayerInLeaderBoard && statGenerator.canLoadData(performanceOverviewProps.playerStats)}
+                    shouldShowLeaderboard={sorted.length > 0 && statGenerator.canLoadData(performanceOverviewProps.playerStats)}
```
Comment (Vexrax) [inline front-end/src/components/player-page/stats/performance-overview.tsx:66]: Whats the reason for this change?  If a LB exists it doesn't mean that the player is in it for sure.

Comment (afieldi) [reply front-end/src/components/player-page/stats/performance-overview.tsx:66]: Yeah but they don't need to be. They can just see where they fall in it

Comment (Vexrax) [reply front-end/src/components/player-page/stats/performance-overview.tsx:66]: That doesnt make sense, we shouldnt be comparing players inter league. a gold player playing in rampage wouldnt have the same stats if they played in dominate.

File: Common/utils.ts (hunk 1 above)
Comment (afieldi) [reply Common/utils.ts:230]: Then we are fucked anyway. Cuz ur += and undefined

File: front-end/src/components/player-page/stats/performance-overview.tsx (hunk 2 above)
Comment (afieldi) [reply front-end/src/components/player-page/stats/performance-overview.tsx:66]: Ok, w.e. It won't make a difference anyway. Reverted.

**Review by Vexrax — APPROVED** [review]
LGTM

**Pull request merged**